libs/langchain/langchain/llms/watsonxllm.py

```diff
@@ -0,0 +1,327 @@
+        """Return type of llm."""
+        return "IBM watsonx.ai"
+
+    def _create_llm_result(self, response: Any) -> LLMResult:
+        """Create the LLMResult from the choices and prompts."""
+        generations = []
+        for res in response:
+            finish_reason = res.get("results")[0].get("stop_reason")
+            gen = Generation(
+                text=res.get("results")[0].get("generated_text"),
+                generation_info={
+                    "generated_token_count": res.get("results")[0].get(
+                        "generated_token_count"
+                    ),
+                    "input_token_count": res.get("results")[0].get("input_token_count"),
+                    "finish_reason": finish_reason,
+                },
+            )
+            generations.append([gen])
+        llm_output = {"model_id": self.model_id}
+        return LLMResult(generations=generations, llm_output=llm_output)
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> str:
+        """Call the IBM watsonx.ai inference endpoint.
+        Args:
+            prompt: The prompt to pass into the model.
+            stop: Optional list of stop words to use when generating.
+            run_manager: Optional callback manager.
+        Returns:
+            The string generated by the model.
+        Example:
+            .. code-block:: python
+
+                response = watsonxllm("What is a molecule")
+        """
+        result = self._generate(
+            prompts=[prompt], stop=stop, run_manager=run_manager, **kwargs
+        )
+        return result.generations[0][0].text
+
+    def _generate(
+        self,
+        prompts: List[str],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        stream: Optional[bool] = None,
+        **kwargs: Any,
+    ) -> LLMResult:
+        """Call the IBM watsonx.ai inference endpoint which then generate the response.
+        Args:
+            prompts: List of strings (prompts) to pass into the model.
+            stop: Optional list of stop words to use when generating.
+            run_manager: Optional callback manager.
+        Returns:
+            The full LLMResult output.
+        Example:
+            .. code-block:: python
+
+                response = watsonxllm.generate(["What is a molecule"])
+        """
+        should_stream = stream if stream is not None else self.streaming
+        if should_stream:
+            generation: Optional[GenerationChunk] = None
+            stream_iter = self._stream(
+                prompts[0], stop=stop, run_manager=run_manager, **kwargs
+            )
+            for chunk in stream_iter:
+                if generation is None:
+                    generation = chunk
+                else:
+                    generation += chunk
```

Comment:
With this approach you lose data about generated tokens, etc. (they will be overwritten). I would rather extract token usage to property `token_usage` and then merge it.